addon/instance-initializers/content-security-policy.js

```diff
@@ -0,0 +1,34 @@
+import { assert } from '@ember/debug';
+
+// reads addon config stored in meta element
+function readAddonConfig(appInstance) {
+  let config = appInstance.resolveRegistration('config:environment');
+  let addonConfig = config['ember-cli-content-security-policy'];
+
+  // TODO: do not require policy to be stored in config object
+  //       if already available through CSP meta element
+  assert(
+    'Required configuration is available at run-time',
+    addonConfig.hasOwnProperty('reportOnly') && addonConfig.hasOwnProperty('policy')
+  );
+
+  return config['ember-cli-content-security-policy'];
+}
+
+export function initialize(appInstance) {
+  let fastboot = appInstance.lookup('service:fastboot');
```

Comment:
Can you wrap the guts of this function in:

```js
if (typeof FastBoot !== 'undefined') {

}
```

That is probably the fastest way to bail out in that circumstance...

Reply:
Are you sure this is needed? I thought putting the instance initializer in `fastboot/` directory ensures that will only be included if app uses FastBoot.